File: templates/todo/api/js/src/models/todoItem.ts
import mongoose, { Schema } from "mongoose";

export enum TodoItemState {
    Todo = "todo",
    InProgress = "inprogress",
    Done = "done"
}

export type TodoItem = {
    id: mongoose.Types.ObjectId
    listId: mongoose.Types.ObjectId
    name: string
    state: TodoItemState
    description?: string
    dueDate?: Date
    completedDate?: Date
    createdDate?: Date
    updatedDate?: Date
}

const schema = new Schema({
    listId: {
        type: Schema.Types.ObjectId,
        required: true
    },
    name: {
        type: String,
        required: true
    },
    description: String,
    state: {
        type: String,
        required: true,
        default: TodoItemState.Todo
    },
    dueDate: Date,
    completedDate: Date,
}, {
    timestamps: {
        createdAt: "createdDate",
        updatedAt: "updatedDate"
    }
});

export const TodoItemModel = mongoose.model<TodoItem>("TodoItem", schema, "TodoItem");

File: templates/todo/api/csharp-cosmos-sql/TodoItem.cs
namespace SimpleTodo.Api;

public class TodoItem
{
    public TodoItem(string listId, string name)
    {
        ListId = listId;
        Name = name;
    }

    public string? Id { get; set; }
    public string ListId { get; set; }
    public string Name { get; set; }
    public string? Description { get; set; }
    public string State { get; set; } = "todo";
    public DateTimeOffset? DueDate { get; set; }
    public DateTimeOffset? CompletedDate { get; set; }
    public DateTimeOffset? CreatedDate { get; set; } = DateTimeOffset.UtcNow;
    public DateTimeOffset? UpdatedDate { get; set; }
}